src/Microsoft.Health.Fhir.Core/Features/Persistence/ResourceWrapperFactory.cs

```diff
@@ -71,6 +72,8 @@ public ResourceWrapper Create(ResourceElement resource, bool deleted, bool keepM
         {
             RawResource rawResource = _rawResourceFactory.Create(resource, keepMeta);
             IReadOnlyCollection<SearchIndexEntry> searchIndices = _searchIndexer.Extract(resource);
```

Comment:
>_searchIndexer.Extract(resource) [](http://example.com/codeflow?start=66&length=32)

would it be worth to put distinct part inside Extract method rather than call it here?

I would assume uniqueness of search indices would be useful in other places as well.

Reply:
Good point, moved distinct to Extract method!